libraries/api/src/middleware/queries.js

```diff
@@ -22,12 +22,14 @@ const queryValidator = ({context, entry}) => {
       }
       if (pattern instanceof RegExp) {
         if (!pattern.test(value)) {
+          delete req.query[key];
           errors.push('Query-string parameter: ' + key + '="' + value +
                       '" does not match expression: ' + pattern.toString());
         }
       } else {
         const msg = pattern.call(context, value);
         if (typeof msg === 'string') {
+          delete req.query[key];
```

Comment:
@djmitche just want to call this out in particular as a more possibly breaking change here to make sure you think it is ok.

Reply:
If I'm reading correctly, this is omitting values from `req.query` in cases where the query was ill-formed, which was already a 400 error and not calling the handler, right?  If so, then I'm not worried.